src/test/java/com/example/bank/userRegister/UserServiceTest.java

```diff
@@ -1,106 +1,139 @@
+        Assertions.assertEquals(USER, userForTest);
+    }
+
+    @Test
+    void shouldReturnIllegalArgumentExceptionWhenAgeIsBelow18() {
+        //given
+        //when
+//        when(userService.registerUser(USER_REQUEST)).thenThrow(new IllegalArgumentException());
+        //then
+        /** Czy może być tylko asersja w poprawnym teście??*/
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            userService.registerUser(USER_BELOW_18_REQUEST);
+        }, "User has to be adult.");
+    }
+
+    @Test
+    void shouldGetUserAndReturnUserFromId3() {
         //given - arrange
-        Map<Long, User> db = new HashMap<>();
         //when - act
-//        when(userRepository.save(USER)).thenReturn(db.put(1L,USER));
-        verify(userRepository.save(USER)).toString();
-//        User newUser = userRepository.save(USER);
-//        when(userRepository.save(USER)).thenReturn(USER);
-//        when(userRepository.findById(USER.getId())).thenReturn(Optional.of(USER));
-//        User user = userService.registerUser(USER_REQUEST);
-        User newUser = userService.registerUser(USER_REQUEST);
-        when(userService.registerUser(USER_REQUEST)).thenReturn(db.put(1L,newUser));
-//        lenient().
-//        User user = verify(userService).registerUser(userRequestForTest);
-//       User myUser = userService.getUser(1L);
+        when(userRepository.findById(USER_WITH_ID3.getId())).thenReturn(Optional.of(USER_WITH_ID3));  //stub
+        userRepository.findById(USER_WITH_ID3.getId());  //act
         //then - assert
-        Assertions.assertEquals(newUser, USER);
-        Assertions.assertEquals(newUser, db.get(1L));
+        verify(userRepository).findById(USER_WITH_ID3.getId());  //verify
     }
 
     @Test
-    void shouldReturnUserFromFirstId() throws UserNotFoundException {
+    void shouldReturnUserExceptionWhenThereIsNoUserWithId() {
         //given
-        Long id = 1L;
+        Long id = 2540L;
         //when
-//        when(userService.getUser(1L)).thenReturn(userForTest);
-//        Optional<User> optionalUser = userRepository.findById(id);
-        when(userRepository.findById(id)).thenReturn(Optional.of(USER));
-        /** Jak mockować takie metody, które mają w sobie inne metody, które odwołują się do db??
-         *  Przecież wywołanie userService.getUser(id) poniżej zawsze zwróci null.
-         *  Jak to testować metody, żeby nie testować, czy Mockito działa?? */
-        User user = userService.getUser(id);
-        //then
-//        assertEquals(testUser, user);
-        Assertions.assertEquals(USER, user);
+        /** Ten test nie działa.
+         * Jeżeli ta metoda zwraca Optionala, czyli NoSuchElementException, to czy w ogóle możemy wymusić
+         * rzucenie własnego wyjątku UserNotFoundException i weryfikację własnego wyjątku w asercji??
+         * Z drugiej strony przez Postmana rzuca UserNotFoundException, więc gdzie jest błąd? */
```

Comment:
to co rzucane jest z postmana to jest to co faktycznie metoda rzuca. Trzeba zobaczyc co metoda robi. Ale asercja powinna weryfikowac faktycznie rzucony wyjatek